include/aspect/parameters.h

```diff
@@ -80,6 +80,21 @@ namespace aspect
       };
     };
 
+    static
+    bool
+    is_defect_correction(const typename NonlinearSolver::Kind &input)
+    {
+      return input == NonlinearSolver::iterated_Advection_and_Newton_Stokes ||
+             input == NonlinearSolver::single_Advection_iterated_Newton_Stokes ||
+             input == NonlinearSolver::no_Advection_iterated_defect_correction_Stokes ||
+             input == NonlinearSolver::single_Advection_iterated_defect_correction_Stokes ||
+             input == NonlinearSolver::iterated_Advection_and_defect_correction_Stokes
```

Comment:
I think the function's name doesn't actually represent what it is doing. That's because a "defect correction" is not the same as a first-order Picard-type iteration. More precisely, "defect" is an older word for "residual", and "correction" means that we compute an *update* from the defect. So DC-Picard definitely falls into this category, but `iterated_advection_and_Newton_Stokes` and `single_Advection_iterated_Newton_Stokes` don't actually fall into the "defect correction" category.

Start with the use cases: Why do the places where you use this function group them together? Maybe we can derive a better name that is based on the *consequences*, not a commonality of the schemes you're considering.

Reply:
hmm, I always considered the Newton solver as a defect correction scheme, where we potentially adjust the update a bit based on the derivatives. The reason for that is that is how it is implemented, and they all follow exactly the same logic and limitations with for example how the boundary conditions are set up. How I view it is that the consequence is that instead of `Ax=b` we now solve `Jdx = Ax-b` where the dx is a update of x and J can be A or the actual Jacobian (or something in between). Would it be more correct to just call it a correction scheme instead of a defect correction scheme, since in both cases we compute a update?

Reply:
I think the question is still why `single_Advection_...` is part of the function, because that `single_Advection` isn't computing an "update" `dx` but computes `x` right away.

As I mentioned before, I think the better approach is to start with the place where you *need* this function and ask what the *property* is that you require.

Reply:
> I think the question is still why `single_Advection_...` is part of the function, because that `single_Advection` isn't computing an "update" `dx` but computes `x` right away.

I would think that the Stokes part can still be nonlinear, so it may have to iterated in a defect correction way for the first timestep. The defect correction is only about the Stokes part.